integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextbest/NextBestQueue.java

```diff
@@ -0,0 +1,98 @@
+package no.difi.meldingsutveksling.nextbest;
+
+import no.difi.meldingsutveksling.IntegrasjonspunktNokkel;
+import no.difi.meldingsutveksling.config.IntegrasjonspunktProperties;
+import no.difi.meldingsutveksling.dokumentpakking.service.CmsUtil;
+import no.difi.meldingsutveksling.dokumentpakking.xml.Payload;
+import no.difi.meldingsutveksling.domain.MeldingsUtvekslingRuntimeException;
+import no.difi.meldingsutveksling.domain.sbdh.EduDocument;
+import no.difi.meldingsutveksling.noarkexchange.MessageException;
+import no.difi.meldingsutveksling.noarkexchange.StatusMessage;
+import org.apache.commons.compress.utils.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.xml.bind.DatatypeConverter;
+import java.io.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+@Component
+public class NextBestQueue {
+
+    private static final Logger log = LoggerFactory.getLogger(NextBestQueue.class);
+
+    @Autowired
+    private IncomingConversationResourceRepository repoIn;
+
+    @Autowired
+    private OutgoingConversationResourceRepository repoOut;
+
+    @Autowired
+    private IntegrasjonspunktNokkel keyInfo;
+
+    @Autowired
+    private IntegrasjonspunktProperties props;
+
+    public void enqueueEduDocument(EduDocument eduDocument) throws IOException, MessageException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, no.difi.meldingsutveksling.noarkexchange.MessageException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1160)